### PR TITLE
Milestone 4

### PR DESCRIPTION
**Pull request by Trictonicmp**
#### In this milestone I:

- [x] Changed the root path to be users list
- [x] Added route for user/id
- [x] Added route for user/id/posts
- [x] Added route for user/id/posts/id

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class UsersController < ApplicationController
+  def index; end
+
+  def show; end
+
+  def posts; end
+
+  def show_post; end
```
Comment (sja-thedude) [inline app/controllers/users_controller.rb:8]: - [x] This is not required for the users controller.

- [x] Kindly make sure to make a file posts_controller and define it in the same way as users controller. Example:
```ruby
class PostsController < ApplicationController
  
  def index; end

  def show; end

end
```

File: app/views/users/show_post.html.erb [hunk 2]
```diff
@@ -0,0 +1 @@
+here is the user <%= params[:id]%> post with id <%= params[:post_id]%>
```
Comment (sja-thedude) [inline app/views/users/show_post.html.erb]: - [x]  Kindly make sure to store this file in # app/views/posts/show.html.erb and rename it as show.html.erb

File: app/views/users/posts.html.erb [hunk 3]
```diff
@@ -0,0 +1 @@
+Here is the user <%= params[:id] %> posts list
```
Comment (sja-thedude) [inline app/views/users/posts.html.erb]: - [x] Kindly make sure to store this file in # app/views/posts/index.html.erb and rename this file as index.html.erb

File: config/routes.rb [hunk 4]
```diff
@@ -2,5 +2,9 @@
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
-  # root "articles#index"
+  root "users#index"
+  resources :users
+  get'/users/:id', to: 'users#show'
+  get'/users/:id/posts', to: 'users#posts'
+  get'users/:id/posts/:post_id', to: 'users#show_post'
```
Comment (sja-thedude) [inline config/routes.rb]: - [x] For the routes, you can simply declare all pages as:
```ruby
 resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @Trictonicmp, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ Professional Readme
✔️ All linters passing
✔️ Correct gitflow

## Required 
- [x] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Trictonicmp** [comment]
Hi @sja-thedude!
Thank you so much for your review and thank you for the info regarding the routes, I just have one question and I know this may be out of scope but, for ```/users/452/posts/25``` how do I access the second id?

**Review by descholar-ceo — APPROVED** [review]
Hi @Trictonicmp 🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Implemented the requested changes ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**